[GSoD 2021] Improved contributing guidelines

iamrajiv opened the pull request:
Fixes: #1111
Related: #1012 wechaty/grpc#140 wechaty/openapi#16

![Screenshot 2021-09-11 at 5 51 28 AM](https://user-images.githubusercontent.com/42106787/132929847-d5fbc5ed-adf9-4c41-9d33-107872a9344a.jpg)

![Screenshot 2021-09-11 at 5 51 36 AM](https://user-images.githubusercontent.com/42106787/132929850-44e9a0d7-3b30-42be-a5e5-fbf90e717924.jpg)

![Screenshot 2021-09-11 at 5 52 05 AM](https://user-images.githubusercontent.com/42106787/132929854-213afeac-08c3-40c0-9b9f-fd98c5924817.jpg)

iamrajiv commented:
@huan @Rohitesh-Kumar-Jain @proudofsimin Please review

huan commented on docusaurus/docs/contributing/documentation.md:
```diff
@@ -46,8 +46,7 @@ Then build the website:
 
-To get started contributing, you'll want to read the :ref:`reStructuredText
-reference <sphinx:rst-index>`.
+To get started contributing, you'll want to read the :ref:`reStructuredText reference <sphinx:rst-index>`.
```
Please remove the RST related information because we are using Markdown.

Rohitesh-Kumar-Jain commented on docusaurus/docs/contributing/overview.md:
```diff
@@ -3,9 +3,22 @@ slug: /contributing/
+As an open-source product, Wechaty thrives on the contributions of community members. Whatever your skill set is, there is a lot you can do to help us make Wechaty better!
+So start forking!
+
+### Not sure where to start?
```
Apart from contributing code/documentation, there are many subtle ways via which one can contribute.

Like reporting bugs, suggesting enhancements, attending open wechaty meetings.
And some more subtle as following Wechaty across social media, and sharing its post

Rohitesh-Kumar-Jain commented:
You haven't attached all the ss

Rohitesh-Kumar-Jain reviewed (commented):
We probably should also add some guidelines about adding relevant labels regarding the PRs and Issues.

Rohitesh-Kumar-Jain commented on docusaurus/docs/contributing/overview.md:
```diff
@@ -70,13 +134,17 @@ Wechaty community and others to maintain a great ecosystem to work in:
 
 We're looking forward to working with you. Welcome aboard! ⛵️
 
+We built wechaty with pleasure because it can help others. Help from you for wechaty will be very much appreciated by the community.
```
`Help from you` -> your contributions (or something else).

I feel that you have used a single word `help` quite often.

Rohitesh-Kumar-Jain commented:
We probably can also add the `Code of Conduct` and `Getting Help` sections in which include a link to the Wechaty Gitter channel as well.

iamrajiv commented:
hi, @huan @Rohitesh-Kumar-Jain thanks for the suggestions I have implemented the requested changes. Also, I have added only those page screenshots that I had heavily edited or created.

huan commented on docusaurus/docs/contributing/overview.md:
```diff
@@ -16,26 +31,56 @@ The work on Wechaty itself falls into three major areas:
 ### Writing documentation ✍️
 
-Wechaty's documentation is one of its key strengths. It's informative
+What's documentation is one of its key strengths? It's informative
```
I feel it's strange in this sentence.

iamrajiv replied:
@huan thanks for pointing out I have fixed it.

proudofsimin commented on docusaurus/docs/contributing/documentation.md:
```diff
@@ -46,8 +58,177 @@ Then build the website:
+  hesitate to refer the reader back to the appropriate tutorial rather than
+  repeat the same material.
+
+## Writing style
+
+When using pronouns in reference to a hypothetical person, such as "a user with
+a session cookie", gender-neutral pronouns (they/their/them) should be used.
+Instead of:
+
+- he or she... use they.
+- him or her... use them.
+- his or her... use their.
+- his or hers... use theirs.
+- himself or herself... use themselves.
+
+Try to avoid using words that minimize the difficulty involved in a task or
+operation, such as "easily", "simply", "just", "merely", "straightforward", and
+so on. People's experience may not match your expectations, and they may become
+frustrated when they do not find a step as "straightforward" or "simple" as it
+is implied to be.
```
If you are documenting this part, suggest either to rewrite the whole content or remove this. It does not reflect the documents we were trying to do in GSoD. Besides, in technical documents, pronouns are actually not encouraged since it might cause some confusions. 
For the modifier, your statement is correct but this is not the main issue. The main focus should be formats, writing tones, and sentence structure. Minor issues such as use of punctuations, spellings...etc.

Last but not least, we are not creating documentation standards ourselves. We follow Google's standards which are very comprehensive in technical writing. As we cannot listed all the rules here, we ought to suggest contributors to check out the rules in Google. Also alternatives are Microsoft's style guide, Red Hat's or Apple's which are very renowned in technical documentation.

proudofsimin commented on docusaurus/docs/contributing/documentation.md:
```diff
@@ -46,8 +58,177 @@ Then build the website:
+frustrated when they do not find a step as "straightforward" or "simple" as it
+is implied to be.
+
+## Commonly used terms
```
I would suggest to merge this part with the styling as Google provides a complete dictionary about whether the words are appropriate and how to format them, but it's thoughtful that you listed down some words.
How do you think rather than listed down in words, we list them down in rules? For example,
1. Use American spellings: "_realize_" instead of "_realise_", "_color_" instead of "_colour_"
2. Use the full product/language names and don't change the letter cases: "_Wechaty_" instead of "_wechaty_", "_JavaScript_" instead of "_javascript_"
...

proudofsimin commented on docusaurus/docs/contributing/documentation.md:
```diff
@@ -46,8 +58,177 @@ Then build the website:
+- **website** &mdash; use one word, without capitalization.
+
+## Wechaty-specific terminology
+
+- **Puppet** &mdash; it's capitalized.
+
+- **Puppet Provider** &mdash; it's capitalized.
+
+- **Puppet Service** &mdash; it's capitalized.
+
+- **TOKEN** &mdash; it's all capitalized.
```
Could refer to [Main concepts in Wechaty](https://wechaty.js.org/docs/main-concepts-in-wechaty).

proudofsimin commented on docusaurus/docs/contributing/documentation.md:
```diff
@@ -46,8 +58,177 @@ Then build the website:
+- **TOKEN** &mdash; it's all capitalized.
+
+## Guidelines for Markdown files
+
```
should also provide the link to official Markdown guides.

proudofsimin reviewed (changes requested):
We could talk about the style more, but as in my previous experience and Google standards, we are actually trying to eliminate using "_you_" in documents to make the tone soften and formal. I understand that each document has its own writing style and those Django used don't necessarily apply to Wechaty's.

Rohitesh-Kumar-Jain reviewed (approved):
Thanks for implementing my suggestion : )

iamrajiv commented:
@proudofsimin @huan I have done the requested changes.

iamrajiv commented:
@huan I am done with the changes can you approve and merge 😀
